operators/video_encoder/video_encoder_request/video_encoder_utils.hpp

```diff
@@ -33,6 +36,39 @@ enum struct EncoderConfig {
   kUnsupported = 3  // Unsupported parameter
 };
 
+static std::unordered_map<std::string, nvidia::gxf::EncoderInputFormat> const
+    EncoderInputFormatMapping = {
+        {"nv12", nvidia::gxf::EncoderInputFormat::kNV12},
+        {"nv24", nvidia::gxf::EncoderInputFormat::kNV24},
+        {"yuv420planar", nvidia::gxf::EncoderInputFormat::kYUV420PLANAR},
+        {"unsupported", nvidia::gxf::EncoderInputFormat::kUnsupported},
+};
```

Comment:
To make it efficient, please consider using `constexpr` for this logic.

```cpp
#include <array>
#include <string_view>
#include <utility>  // for std::pair

namespace nvidia::gxf {

enum class EncoderInputFormat {
    kNV12,
    kNV24,
    kYUV420PLANAR,
    kUnsupported,
};

constexpr std::array<std::pair<std::string_view, nvidia::gxf::EncoderInputFormat>, 4> EncoderInputFormatMapping = {{
    {"nv12", nvidia::gxf::EncoderInputFormat::kNV12},
    {"nv24", nvidia::gxf::EncoderInputFormat::kNV24},
    {"yuv420planar", nvidia::gxf::EncoderInputFormat::kYUV420PLANAR},
    {"unsupported", nvidia::gxf::EncoderInputFormat::kUnsupported},
}};

constexpr EncoderInputFormat get_encoder_input_format(std::string_view format) {
    for (const auto& [key, value] : EncoderInputFormatMapping) {
        if (key == format) {
            return value;
        }
    }
    return nvidia::gxf::EncoderInputFormat::kUnsupported;
}

}  // namespace nvidia::gxf

int main() {
    using namespace nvidia::gxf;
    constexpr auto format = get_encoder_input_format("gv12");
    static_assert(format == EncoderInputFormat::kGV12, "Format should be kGV12");

    return 0;
}
```

Same for `EncoderConfigMapping`.

Reply:
Thanks for the tip!